### PR TITLE
Set 159

### PR DESCRIPTION
**Pull request by PeterAlabaster**

### CODE REVIEW

File: addons/udes_stock_cron/models/stock_picking.py [hunk 1]
```diff
@@ -162,21 +167,30 @@ def raise_insufficiency_error(moves):
...
+                """Abstract iteration over fetched pickings."""
+                while True:
+                    if self:
+                        # Removed processed pickings from self
+                        yield self.filtered(lambda p: p.picking_type_id == picking_type) - processed
+                    else:
+                        domain = base_domain[:]
+                        if processed:
+                            domain.append(("id", "not in", processed.ids))
+                        pickings = Picking.search(domain, limit=batch_size)
```
Comment (jrbirch99) [inline addons/udes_stock_cron/models/stock_picking.py]: nit: This should be `limit=limit` I think, but they look interchangeable

**Review by adamrpatrick — APPROVED** [review]

File: addons/udes_stock_cron/models/stock_picking.py [hunk 2]
```diff
@@ -1,20 +1,14 @@
-import time
-from psycopg2 import OperationalError, errorcodes
 from collections import defaultdict
 
 from odoo import models, _
 from odoo.exceptions import UserError
+from odoo.addons.udes_common.tools import odoo_retry
```
[the remark this answers is not in the document]
Comment (PeterAlabaster) [reply addons/udes_stock_cron/models/stock_picking.py:5]: Will get them reordered

File: addons/udes_stock_cron/models/stock_picking.py (hunk 1 above)
Comment (PeterAlabaster) [reply addons/udes_stock_cron/models/stock_picking.py]: not sure how this ended up showing in the diff, this looks to be before i made any changes :confused: `limit` param should be `batch_size` as there is no `limit` variable defined any more

File: addons/udes_stock_cron/models/stock_picking.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jrbirch99) [reply addons/udes_stock_cron/models/stock_picking.py:5]: They look re-ordered so I will resolve

**Review by jrbirch99 — APPROVED** [review]

File: addons/udes_stock_cron/models/stock_picking.py (hunk 1 above)
Comment (PeterAlabaster) [reply addons/udes_stock_cron/models/stock_picking.py]: I thought i did that all in one commit :thinking: i'll go back and comb over them to make sure no changes got accidentally changes in other commits

Comment (PeterAlabaster) [reply addons/udes_stock_cron/models/stock_picking.py]: Ah, it was one of the ports from 11.0 improvements that changed it. Completely forgot about that!

**Pull request merged**